geowebcache/core/src/main/java/org/geowebcache/storage/UnsuitableStorageException.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.geowebcache.storage;
+
+/**
+ * Thrown when a Blobstore was initialized with persistance that is unsuitable, such as a non-empty
+ * directory
+ * @author smithkm
+ *
+ */
+public class UnsuitableStorageException extends StorageException {
+    
+    public UnsuitableStorageException(String msg) {
+        super(msg);
+    }
+
+    public static void checkSuitability(String location, final boolean exists, boolean empty)
```

Comment:
Its a bit weird for an exception to contain a utility method (even if that method does throw said exception), this seems like it would be more appropriate in CompositeBlobStore, same as where you define the thread local being tested